pkg/test/assert.go

```diff
@@ -104,11 +104,16 @@ func Errors(namespace string, timeout int, errorFiles ...string) error {
 	return errors.New("error asserts not valid")
 }
 
-func Client(_ bool) (client.Client, error) {
+func Client(_ bool, as string) (client.Client, error) {
 	cfg, err := config.GetConfig()
```

Comment:
How about:
- extract this and the next block (and an identical piece in `DiscoveryClient` below) into a new function in a new package
- in that package create an exported package-level variable `As` (or perhaps a bit more meaningful name) that defaults to an empty string.
- in that extracted code, use this `As` variable to tweak the config
- in the root cobra command, add the flag that sets the value of this new variable (I hope it's possible 🤔 )

Reply:
@porridge the method you have described is possible and much cleaner if you are happy with the appraoch (see updated) code I will roll this out to the other commands that generate a client config.

Reply:
Sorry, perhaps I should have been more precise - when saying "this and the next block" above I meant just the following piece, not the whole `GetClient ` function:

```
	cfg, err := config.GetConfig()
	if err != nil {
		return nil, err
	}

	if ImpersonateAs != "" {
		cfg.Impersonate.UserName = ImpersonateAs
		}
```

As for the name of the package, perhaps something like `pkg/k8s` such that when imported as `client` the call would look like `k8s. Config()`? 🤔  WDYT?

Reply:
But this looks very promising, thank you for wishing on this!

Reply:
Yeah I agree, we can also nest other k8s utilities under this package. I have moved the config generation to a function in this package.